pkg/operator/ceph/cluster/nodedaemon/exporter.go

```diff
@@ -0,0 +1,260 @@
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+
+	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	monitoringPath                   = "/etc/ceph-monitoring/"
+	serviceMonitorFile               = "exporter-service-monitor.yaml"
+	sockDir                          = "/run/ceph"
+	perfCountersPrioLimit            = "5"
+	statsPeriod                      = "5"
+	DefaultMetricsPort        uint16 = 9926
+	exporterServiceMetricName        = "ceph-exporter-http-metrics"
+)
+
+// createOrUpdateCephExporter is a wrapper around controllerutil.CreateOrUpdate
+func (r *ReconcileNode) createOrUpdateCephExporter(node corev1.Node, tolerations []corev1.Toleration, cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion) (controllerutil.OperationResult, error) {
+	if !cephVersion.IsAtLeast(cephver.CephVersion{Major: 17, Minor: 2, Extra: 5}) {
+		logger.Infof("Skipping exporter reconcile on ceph version %q", cephVersion.String())
+		return controllerutil.OperationResultNone, nil
+	}
+
+	// Create or Update the deployment default/foo
+	nodeHostnameLabel, ok := node.ObjectMeta.Labels[corev1.LabelHostname]
+	if !ok {
+		return controllerutil.OperationResultNone, errors.Errorf("label key %q does not exist on node %q", corev1.LabelHostname, node.GetName())
+	}
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      k8sutil.TruncateNodeName(fmt.Sprintf("%s-%%s", cephExporterAppName), nodeHostnameLabel),
+			Namespace: cephCluster.GetNamespace(),
+		},
+	}
+	err := controllerutil.SetControllerReference(&cephCluster, deploy, r.scheme)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Errorf("failed to set owner reference of ceph-exporter deployment %q", deploy.Name)
+	}
+
+	configDir := path.Join(cephCluster.Spec.DataDirHostPath, cephCluster.Namespace)
+	configHostPathType := v1.HostPathDirectory
+	src := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: configDir, Type: &configHostPathType}}
+	volumes := append(controller.DaemonVolumesBase(config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath), "", cephCluster.Spec.DataDirHostPath), v1.Volume{Name: "ceph-conf-dir", VolumeSource: src})
+
+	mutateFunc := func() error {
+
+		// labels for the pod, the deployment, and the deploymentSelector
+		deploymentLabels := map[string]string{
+			corev1.LabelHostname: nodeHostnameLabel,
+			k8sutil.AppAttr:      cephExporterAppName,
+			NodeNameLabel:        node.GetName(),
+		}
+		deploymentLabels[config.CephExporterType] = "ceph-exporter"
```

Comment:
These three labels (lines 87-89) don't really add value, let's remove them. I know the crash collector has them, but they aren't needed either.